src/vdom.rs

```diff
@@ -138,71 +174,50 @@ impl VDom<Node> {
 
     pub(crate) fn step(&mut self, host: &Host, input: VDom<()>) {
         match input.vdom {
-            VDomNode::Text { text: tnew, .. } => match &mut self.vdom {
-                VDomNode::Text {
-                    text: told,
-                    state: node,
-                } => {
-                    if tnew != *told {
-                        node.set_text_content(&tnew);
-                        *told = tnew;
+            VDomNode::Text(mut new) => match self.vdom {
+                VDomNode::Text(mut old) => {
+                    let mut node = old.state;
+                    if new.text != old.text {
+                        node.set_text_content(&new.text);
+                        old.update(Text::new(new.text.clone(), old.state));
                     }
                 }
-                VDomNode::Elem { .. } => {
-                    self.halt_and_build(
-                        host,
-                        VDom {
-                            vdom: VDomNode::Text {
-                                text: tnew,
-                                state: (),
-                            },
-                        },
-                    );
+                VDomNode::Elem(elem) => {
+                    let node = VDomNode::Text(Text {
```

Comment:
We should use the `new` function for VDomNode here (after it has been made to work with any state).